exchange/replication_test.go

```diff
@@ -532,6 +560,7 @@ func TestMultiDispatchStreams(t *testing.T) {
 		p := tnds[r.Provider]
 		p.VerifyFileTransferred(ctx, t, p.DAG, rootCid, origBytes)
```

Comment:
Can you verify that hn1 did not actually receive the content?

Reply:
I'm struggling to verify that, I know it did not receive it thanks to this line : https://github.com/myelnet/pop/blob/main/exchange/replication.go#L384 but how can I make sure it hasn't received it when it already has it? Can I have a hint? 😁

Reply:
btw I don't think it's useful to log the "payload CID already exists", I can remove it or setting it as Debug() maybe?

Reply:
> I'm struggling to verify that, I know it did not receive it thanks to this line : https://github.com/myelnet/pop/blob/main/exchange/replication.go#L384 but how can I make sure it hasn't received it when it already has it? Can I have a hint? 😁

Actually I just realized you're already verifying by checking `tnds[r.Provider]` if the provider has hn1 it wouldn't be in that map.

Reply:
> btw I don't think it's useful to log the "payload CID already exists", I can remove it or setting it as Debug() maybe?

What log is that?

Reply:
> > btw I don't think it's useful to log the "payload CID already exists", I can remove it or setting it as Debug() maybe?
> 
> What log is that?

https://github.com/myelnet/pop/blob/main/exchange/replication.go#L382

Reply:
Sure

Reply:
Ok if think I understand the replication process, based on TestMultiDispatchStreams :
- we create one node (n1) that has a new file (f1)
- n1 is connected to 7 neighbours : 6 without f1 + 1 with f1 already (I made it for this PR)
- n1 will tell its 7 neighbours it has a new file f1, and expects that 6 of them will ask for retrieving of f1 ( 6 = Replication Factor)
- the 6 neighbours will then ask for f1 retrieval and the 7th will not because he already has it
- n1 sends the f1 to all of them and once n1 receives the Completed datatransfer.Event, it sends a PRecord to its "out" channel
- in our test we make sure we received 6 PRecords and that all these 6 peers received the whole file

But for me, we haven't really tested that the 7th neighbour didn't ask for the content of f1, so why not using 6 neighbours instead of 7, or increasing RF to 7, so that we make sure that the 6 peers receive the file, and then we expect a timeout for the 7th that already has the file, so that we are sure he didn't ask for it? It might seem hacky so idk

Reply:
I mentioned above: you're alright because the test iterates over the 6 records and makes sure the 7th peer that already has the content isn't part of those records. To verify that the test would break if the 7th peer were to retrieve the content just remove the index check and it should error out (not always as peers are selected randomly)